### PR TITLE
Feat(eos_cli_config_gen): Implement NAT profiles + ethernet interfaces CLI

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

* Re-using the template from interface for profiles as it works fine (tested on cEOS)
* Not re-using the template for documentation as the tables were not appropriate
* Added the config for `ip nat service-policy <NAME>` 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

TO UPDATE

## How to test
molecule

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-nat.j2 [hunk 1]
```diff
@@ -13,6 +13,120 @@
...
+{%                 for row in ip_nat.src_dyn %}
+| {{ row.acl }} | {{ row.type }} | {{ row.pool }} | {{ row.priority }} | {{ row.comment }} |
+{%                 endfor %}
+{%             endif %}
+{%             if ip_nat.dst_static %}
+
+##### IP NAT: Destination Static
+
+| Direction | Original IP | Original Port | Access List | Translated IP | Translated Port | Protocol | Group | Priority | Comment |
+| --------- | --------- | ----------- | ------------- | ----------- | ------------- | --------------- | -------- | ----- | -------- | ------- |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-nat.j2]:
```suggestion
| Direction | Original IP | Original Port | Access List | Translated IP | Translated Port | Protocol | Group | Priority | Comment |
| --------- | ----------- | ------------- | ----------- | ------------- | --------------- | -------- | ----- | -------- | ------- |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/interfaces-ip-nat.j2 [hunk 2]
```diff
@@ -112,3 +118,13 @@
 | {{ row.if_name }} | {{ row.acl }} | {{ row.pool }} | {{ row.priority }} | {{ row.comment }} |
 {%     endfor %}
 {% endif %}
+{% if ip_nat.profile %}
+
+##### IP NAT: Interfaces configured via profile
+
+| Interface | Profile |
+| --------- | ----------- |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/interfaces-ip-nat.j2]:
```suggestion
| --------- | ------- |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_nat.schema.yml [hunk 3]
```diff
@@ -16,6 +16,18 @@ keys:
...
+        primary_key: name
+        items:
+          type: dict
+          $ref: "eos_cli_config_gen#/$defs/interface_ip_nat"
+          keys:
+            name:
+              type: str
+            vrf:
+              type: str
+              description: Specify VRF for NAT profile.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_nat.schema.yml]:
```suggestion
            vrf:
              type: str
              convert_types:
                - int
              description: Specify VRF for NAT profile.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml [hunk 4]
```diff
@@ -3394,6 +3398,18 @@ keys:
...
+        primary_key: name
+        items:
+          type: dict
+          $ref: eos_cli_config_gen#/$defs/interface_ip_nat
+          keys:
+            name:
+              type: str
+            vrf:
+              type: str
+              description: Specify VRF for NAT profile.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]:
```suggestion
            vrf:
              type: str
              convert_types:
                - int
              description: Specify VRF for NAT profile.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-nat.j2 (hunk 1 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ip-nat.j2]: [6b1fcc9](https://github.com/aristanetworks/ansible-avd/pull/3294/commits/6b1fcc91ad51079a84a3f05fc71c32d3c32663e2)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/interfaces-ip-nat.j2 (hunk 2 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/interfaces-ip-nat.j2]: [6b1fcc9](https://github.com/aristanetworks/ansible-avd/pull/3294/commits/6b1fcc91ad51079a84a3f05fc71c32d3c32663e2)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_nat.schema.yml (hunk 3 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ip_nat.schema.yml]: [6b1fcc9](https://github.com/aristanetworks/ansible-avd/pull/3294/commits/6b1fcc91ad51079a84a3f05fc71c32d3c32663e2)

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml (hunk 4 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: [6b1fcc9](https://github.com/aristanetworks/ansible-avd/pull/3294/commits/6b1fcc91ad51079a84a3f05fc71c32d3c32663e2)

**Review by ClausHolbechArista — APPROVED** [review]

**Review by carlbuchmann — APPROVED** [review]
LGTM

- test configs on dut
- reviewed schema

**Pull request merged**